src/module.h

```diff
@@ -0,0 +1,18 @@
+#ifndef _MODULE_H_
+#define _MODULE_H_
+
+/* This header file exposes a set of functions defined in module.c that are
+ * not part of the module API, but are used by the core to interact with modules
+ */
```

Comment:
I noticed (while reviewing another PR) that all the module structs and prototypes are in `server.h`. We should keep them in one place, so either move these to `server.h` or move everything module related from `server.h` to `module.h`. (The former seems easier.)

Reply:
It's historic. Once upon a time, all the C code was all in one file too, redis.c or something. It was broken up into multiple files, but not the header file.

We can break it into pieces too, but let's do it in a separate PR or issue. There are probably a spaghetti dependencies to sort out.

Reply:
Right, that split can be done in a different PR. I would like to move all module related declarations to `module.h` in a different PR.